app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -182,62 +183,85 @@ class SupplementalApplicationPage extends React.Component {
+    this.setState({ rentalAssistanceLoading: true })
     if (rentalAssistance.type_of_assistance !== 'Other') {
       rentalAssistance.other_assistance_name = null
     }
 
-    // TODO: Add a real call to Salesforce to update the
-    // rental assistance record. Here, we mock the real
-    // Salesforce update by just returning the rental
-    // assistance values.
-    const updatedRentalAssistance = ((rentalAssistance) => rentalAssistance)(rentalAssistance)
+    const response = await apiService.updateRentalAssistance(
+      rentalAssistance,
+      this.state.persistedApplication.id
+    )
 
-    this.setState(prev => {
-      const rentalAssistances = cloneDeep(prev.rentalAssistances)
-      const idx = findIndex(rentalAssistances, { id: updatedRentalAssistance.id })
-      rentalAssistances[idx] = updatedRentalAssistance
-      return {
-        rentalAssistances: rentalAssistances,
-        showNewRentalAssistancePanel: false,
-        showAddRentalAssistanceBtn: true
-      }
-    })
+    if (response) {
+      this.setState(prev => {
+        const rentalAssistances = cloneDeep(prev.rentalAssistances)
+        const idx = findIndex(rentalAssistances, { id: rentalAssistance.id })
+        rentalAssistances[idx] = rentalAssistance
+
+        return {
+          rentalAssistances: rentalAssistances,
+          showNewRentalAssistancePanel: false,
+          showAddRentalAssistanceBtn: true,
+          rentalAssistanceLoading: false
+        }
+      })
+      return true
+    } else {
+      Alerts.error()
+      this.setState({ rentalAssistanceLoading: false })
+      return false
+    }
   }
 
-  handleDeleteRentalAssistance = (rentalAssistance) => {
-    // TODO: Add a real call to Salesforce to delete the
-    // rental assistance record.
-
-    this.setState(prev => {
-      const rentalAssistances = cloneDeep(prev.rentalAssistances)
-      const idx = findIndex(rentalAssistances, { id: rentalAssistance.id })
-      rentalAssistances.splice(idx, 1)
-      return {
-        rentalAssistances: rentalAssistances,
-        showNewRentalAssistancePanel: false,
-        showAddRentalAssistanceBtn: true
-      }
-    })
+  handleDeleteRentalAssistance = async (rentalAssistance) => {
+    this.setState({ rentalAssistanceLoading: true })
+    const response = await apiService.deleteRentalAssistance(rentalAssistance.id)
+
+    if (response) {
+      this.setState(prev => {
+        const rentalAssistances = cloneDeep(prev.rentalAssistances)
+        const idx = findIndex(rentalAssistances, { id: rentalAssistance.id })
+        rentalAssistances.splice(idx, 1)
+        return {
+          rentalAssistances: rentalAssistances,
+          showNewRentalAssistancePanel: false,
+          showAddRentalAssistanceBtn: true,
+          rentalAssistanceLoading: false
+        }
+      })
+      return true
+    } else {
+      Alerts.error()
+      this.setState({ rentalAssistanceLoading: false })
+      return false
+    }
```

Comment:
do you think that it'd be possible to condense the create/update/destroy or are there enough slight differences that it's not worth it?

Reply:
there are enough differences in the post-request steps that it's probably not worth squishing them into a single method; however, perhaps the loading state management and error alert could be DRYed a bit - @fddayan what are your thoughts?

Reply:
I would keep a method for each action. I agree we would DRY up those methods.

Reply:
@akegan Fed has added a commit showing a way to draw out the common parts of the rental assistance action functions - please review and let us know if you'd like to use that new way or keep it as it was before.

Reply:
Ok let's leave it as it is now